photutils/psf/models.py

```diff
@@ -12,8 +12,274 @@
+        self._ndata = data / tflux
+
+        # set input image related parameters:
+        self._ny, self._nx = data.shape
+
+        # find origin of the coordinate system in image's pixel grid:
+        self.origin = origin
+
+        super(Discrete2DModel, self).__init__(flux, x_0, y_0)
+
+        # define interpolating spline:
+        self.compute_interpolator(kwargs)
+
+    @property
+    def ndata(self):
+        """Normalized model data such that sum of all pixels is 1."""
+        return self._ndata
+
+    @property
+    def data(self):
+        """Model data such that sum of all pixels is equal to flux."""
+        return (self.flux.value * self._ndata)
+
+    @property
+    def weights(self):
+        """
+        Weights of image data. When setting weights, :py:class:`numpy.ndarray`
+        or `None` may be used.
+        """
+        return self._weights
+
+    @weights.setter
+    def weights(self, weights):
+        self._weights = weights
+
+    @property
+    def shape(self):
+        """A tuple of dimensions of the data array in numpy style (ny, nx)."""
+        return self._ndata.shape
+
+    @property
+    def nx(self):
+        """Number of columns in the data array."""
+        return self._nx
+
+    @property
+    def ny(self):
+        """Number of rows in the data array."""
+        return self._ny
+
+    @property
+    def origin(self):
+        """
+        A tuple of ``x`` and ``y`` coordinates of the origin of the coordinate
+        system in terms of pixels of model's image.
+
+        When setting the coordinate system origin, a tuple of two `int` or
+        `float` may be used. If origin is set to `None`, the origin of the
+        coordinate system will be set to the middle of the data array
+        (``(npix-1)/2.0``).
+
+        .. warning::
+            Modifying `origin` will not adjust (modify) model's parameters
+            `x_0` and `y_0`.
+        """
+        return (self._ox, self._oy)
+
+    @origin.setter
+    def origin(self, origin):
+        if origin is None:
+            self._ox = (self._nx - 1) / 2.0
+            self._oy = (self._ny - 1) / 2.0
+        elif hasattr(origin, '__iter__') and len(origin) == 2:
+            self._ox, self._oy = origin
+        else:
+            raise TypeError("Parameter 'origin' must be either None or an "
+                            "iterable with two elements.")
+
+    @property
+    def ox(self):
+        """X-coordinate of the origin of the coordinate system."""
+        return self._ox
+
+    @property
+    def oy(self):
+        """Y-coordinate of the origin of the coordinate system."""
+        return self._oy
+
+    @property
+    def fillval(self):
+        """Fill value to be returned for coordinates outside of the domain of
+        definition of the interpolator.
+
+        """
+        return self._fillval
+
+    @fillval.setter
+    def fillval(self, fillval):
+        self._fillval = fillval
+
+    def recenter(self):
```

Comment:
I'm a bit confused why this method exists - it doesn't seem to get used anywhere...?  Or is it used in the other psf builder?